sql/sql_table.cc

```diff
@@ -3657,6 +3657,18 @@ mysql_prepare_create_table(THD *thd, HA_CREATE_INFO *create_info,
     if (sql_field->vcol_info)
       sql_field->flags&= ~NOT_NULL_FLAG;
 
+    if (sql_field->real_field_type() == MYSQL_TYPE_BLOB &&
+        thd->lex->sql_command == SQLCOM_CREATE_TABLE)
```

Comment:
Isn't SQLCOM_CREATE_TABLE always set here?

Why do we need to check for real_field_type? Isn't the type_handler() check enough?

Reply:
Yes both checks are needed, because function `mysql_prepare_create_table` is used for `alter` too.